Apps/GatewayApi/src/Services/DependentService.cs

```diff
@@ -192,6 +193,38 @@ public RequestResult<IEnumerable<DependentModel>> GetDependents(string hdId, int
             return result;
         }
 
+        /// <inheritdoc/>
+        public RequestResult<IEnumerable<GetDependentResponse>> GetDependents(DateTime fromDateUtc, DateTime? toDateUtc, int page = 0, int pageSize = 5000)
+        {
+            // Page size max is 5000
+            pageSize = pageSize <= 5000 ? pageSize : 5000;
+
+            // Get Dependents from database
+            int offset = page * pageSize;
+            DbResult<IEnumerable<ResourceDelegate>> dbResourceDelegates = this.resourceDelegateDelegate.Get(
+                fromDateUtc,
+                toDateUtc,
+                offset,
+                pageSize);
+
+            if (dbResourceDelegates.Status != DbStatusCode.Read)
+            {
+                return RequestResultFactory.ServiceError<IEnumerable<GetDependentResponse>>(ErrorType.CommunicationInternal, ServiceType.Database, dbResourceDelegates.Message);
+            }
+
+            IDictionary<string, IEnumerable<ResourceDelegate>> resourceDelegatesByProfileId =
+                dbResourceDelegates.Payload.GroupBy(g => g.ProfileHdid).ToDictionary(g => g.Key, g => g.AsEnumerable());
+            IEnumerable<GetDependentResponse> dependentResponses = resourceDelegatesByProfileId.Select(
+                    keyValuePair => new GetDependentResponse
+                    {
+                        DelegateId = keyValuePair.Key,
+                        DependentRecords = this.autoMapper.Map<IEnumerable<DependentRecord>>(keyValuePair.Value),
+                    })
+                .ToList();
```

Comment:
I think it may be possible to eliminate the ToDictionary() call.

```
List<GetDependentResponse> resourceDelegatesByProfileId =
    dbResourceDelegates.Payload
        .GroupBy(g => g.ProfileHdid)
        .Select(
            g => new GetDependentResponse
            {
                DelegateId = g.Key,
                DependentRecords = this.autoMapper.Map<IEnumerable<DependentRecord>>(g),
            })
        .ToList();
```